index.js

```diff
@@ -47,9 +49,17 @@ function translate(text, opts) {
         };
         data[token.name] = token.value;
 
-        return url + '?' + querystring.stringify(data);
+        var fullUrl = url + '?' + querystring.stringify(data);
+        if (fullUrl.length > MAX_URL_LENGTH) {
+            delete data.q;
+            return [
+                url + '?' + querystring.stringify(data),
+                {method: 'POST', body: {q: text}}
+            ];
+        }
+        return [fullUrl];
     }).then(function (url) {
-        return got(url).then(function (res) {
+        return got(...url).then(function (res) {
```

Comment:
I suggest to rename `url` -> `params` to be more clear.
```js
then(function (params) {
  return got(params).then(...)
```
And upper:
```js
	        if (fullUrl.length > MAX_URL_LENGTH) {
            delete data.q;
            return {
                url: url + '?' + querystring.stringify(data),
                method: 'POST', 
                body: {q: text}
           };
        }
        return {
          url: fullUrl,
          method: 'GET'
        };

Reply:
This has been implemented in the current commit.